### DISCUSSION TITLE
Imageflow vs libvips

### DISCUSSION THREAD

**Discussion by hanssonrickard** [libvips/libvips#3328] [General]
Hi.
I have been using libvips for some projects now and it is really fast.
But i then saw this page and wonders how it is possible to be even faster (as they claim).
ny input on that? Not that i am interested in using imageflow, it is just the numbers that was interesting.

https://www.imageflow.io/benchmarks/

**Comment by jcupitt** [comment]
Hi @hanssonrickard,

imageflow is nice, but they are comparing against an unoptimised path in libvips.

Background: the imageflow authors are very strong believers in linear-light downsampling. imagemagick have some stuff about this in their docs, if you're not familiar with it:

https://imagemagick.org/Usage/resize/#resize_colorspace

imageflow funded a special linear light downsample mode in libjpeg-turbo:

https://github.com/libjpeg-turbo/libjpeg-turbo/issues/63

But I'm not sure I like it much, and it won't work with other jpeg decoders, like mozjpeg, so libvips has not used this. Instead, `vipsthumbnail` has a `--linear` switch which disables shrink-on-load, does a full conversion to float scRGB, downsamples in that, then converts back to sRGB again. This is obviously pretty slow compared to having it integrated in the decode library, and that's where their 4x faster comes from.

Summary: imageflow is much faster than libvips at **linear light downsampling of JPEG images**, and that's the case they benchmark on their site. 

If you don't do linear light downsampling (almost no one does, it's a very niche thing), or if you are downsampling other formats (eg. PNG etc.), then libvips is quicker. If you are doing complex processing and have many cores available, then libvips can be a lot quicker, since it has a fancier threading system.

So, in practice, libvips is probably faster for most people.

**Reply by hanssonrickard** [reply]
Thanks for the reply and information given.
